### PR TITLE
Add debug tooling to deploy-doc-tests/run

### PR DESCRIPTION
**Pull request by moretea**
Add `-d` and `-s` options to `deploy-doc-tests/run` command

If either of these options is invoked, no test is run.

- `-d`, or `--shell-in-container` starts an interactive docker container with the same 'environment' as where the tests are run.
- `-s`, or `--output-shell-script` does what it's name implies; write a
shell script containing all the steps (in the correct order) to the
standard output.

### CODE REVIEW

File: deploy-doc-tests/lib/deploy_doc_test/plan.rb [hunk 1]
```diff
@@ -58,6 +58,20 @@ def missing_env_vars
       @required_env_vars.select { |e| ENV[e].nil? }
     end
 
+    def to_shell_script
+      lines = ["#!/usr/bin/env bash"]
+      PHASES.each do |phase_name|
+        lines << ("#" * 80)
+        lines << "# Phase #{phase_name}"
+        @steps_in_phases[phase_name].each do |step|
+          p step
```
Comment (vlal-zz) [inline deploy-doc-tests/lib/deploy_doc_test/plan.rb]: does this print the step as well?

**Review by vlal-zz — COMMENTED** [review]
LGTM, just one comment

**Review by frankscholten — COMMENTED** [review]
How about renaming `run` to `deploydoc`?

**Review by pidster — CHANGES_REQUESTED** [review]
Please change 'run' to 'deploydoc' per Frank.

**Comment by moretea** [comment]
I don't think `deploydoc` is the correct term. The invocation would be `deploy-doc-tests/deploy-doc`.
I still think that `deploy-doc-test/run` is better.

A compromise would be to rename it to `deploy-doc-test/run-deploy-doc-test`

**Comment by moretea** [comment]
OK, had a chat with Frank. We consider extracting this as a separate project, having a `deploydoc` command makes sense in that case.

**Review by pidster — DISMISSED** [review]

**Review by pidster — CHANGES_REQUESTED** [review]
There's a merge conflict in `deploy-doc-tests/Dockerfile`.

**Comment by moretea** [comment]
Yes, don't know what happened here, I'm going to investigate the history of this branch a bit.

**Comment by pidster** [comment]
@moretea @frankscholten Making the doc tool a separate project is good idea. If you can't create a repo in this org, shout & I'll sort it for you.

**Comment by moretea** [comment]
Closing, will merge these changes into https://github.com/microservices-demo/deploydoc

**Pull request closed without merging**